publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/persistence/ExpandedDataEntriesPersistenceHandler.java

```diff
@@ -6,66 +6,120 @@
 import no.unit.nva.events.handlers.DestinationsEventBridgeEventHandler;
 import no.unit.nva.events.models.AwsEventBridgeDetail;
 import no.unit.nva.events.models.AwsEventBridgeEvent;
 import no.unit.nva.events.models.EventReference;
 import no.unit.nva.expansion.model.ExpandedDataEntry;
+import no.unit.nva.expansion.model.ExpandedPublishingRequest;
 import no.unit.nva.publication.events.handlers.PublicationEventsConfig;
+import no.unit.nva.publication.events.handlers.tickets.identityservice.CustomerDto;
+import no.unit.nva.publication.external.services.AuthorizedBackendUriRetriever;
+import no.unit.nva.publication.external.services.RawContentRetriever;
 import no.unit.nva.s3.S3Driver;
 import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Failure;
 import nva.commons.core.paths.UnixPath;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ExpandedDataEntriesPersistenceHandler
     extends DestinationsEventBridgeEventHandler<EventReference, EventReference> {
-    
+
     public static final String EXPANDED_ENTRY_PERSISTED_EVENT_TOPIC = "PublicationService.ExpandedEntry.Persisted";
+    public static final EventReference EMPTY_EVENT = new EventReference("Empty event", null);
     private static final Logger logger = LoggerFactory.getLogger(ExpandedDataEntriesPersistenceHandler.class);
+    public static final String APPLICAITON_LD_JSON = "applicaiton/ld+json";
+    public static final String COULD_NOT_RETRIEVE_CUSTOMER = "could not retreive customer";
     private final S3Driver s3Reader;
     private final S3Driver s3Writer;
-    
+
+    private final RawContentRetriever backendClient;
+
     @JacocoGenerated
     public ExpandedDataEntriesPersistenceHandler() {
-        this(new S3Driver(PublicationEventsConfig.EVENTS_BUCKET), new S3Driver(PERSISTED_ENTRIES_BUCKET));
+        this(new S3Driver(PublicationEventsConfig.EVENTS_BUCKET), new S3Driver(PERSISTED_ENTRIES_BUCKET),
+             new AuthorizedBackendUriRetriever("", ""));
     }
-    
-    public ExpandedDataEntriesPersistenceHandler(S3Driver s3Reader, S3Driver s3Writer) {
+
+    public ExpandedDataEntriesPersistenceHandler(S3Driver s3Reader, S3Driver s3Writer,
+                                                 RawContentRetriever backendClient) {
         super(EventReference.class);
         this.s3Reader = s3Reader;
         this.s3Writer = s3Writer;
+        this.backendClient = backendClient;
     }
-    
+
     @Override
     protected EventReference processInputPayload(
         EventReference input,
         AwsEventBridgeEvent<AwsEventBridgeDetail<EventReference>> event,
         Context context) {
         ExpandedDataEntry expandedResourceUpdate = readEvent(input);
+        EventReference outputEvent =
+            shouldBePersisted(expandedResourceUpdate)
+                ? createOutPutEventAndPersistDocument(expandedResourceUpdate)
+                : EMPTY_EVENT;
+        logger.info(outputEvent.toJsonString());
```

Comment:
Ønsker vi å logge empty_event? 
Hvis ikke, kan vi flytte logger.info inn i funksjonen med den faktiske eventen og returnere med en gang

Reply:
Ja, var usikker på det selv. Med empty event så er det tydelig at det som kommer ut vil ingen subscribe på, men man betaler jo for ekstra logging. Kan fjerne.

Reply:
Du får bestemme selv, tenker bare at det ser litt bedre ut rent visuelt.

Reply:
fikset